roles/network_plugin/weave/tasks/pre-upgrade.yml

```diff
@@ -0,0 +1,39 @@
+- name: Weave pre-upgrade | Stop legacy weave
+  command: weave stop
+  failed_when: false
+
+- name: Weave pre-upgrade | Stop legacy systemd weave services
+  service:
+    name: "{{ item }}"
+    enabled: no
+    state: stopped
+  with_items:
+    - weaveexpose
+    - weaveproxy
+    - weave
+  failed_when: false
+
+- name: Weave pre-upgrade | Purge legacy systemd weave systemd unit files
+  file:
+    path: "{{ item }}"
+    state: absent
+  register: purged_weave_systemd_units
+  with_items:
+    - "/etc/systemd/system/weaveexpose.service"
+    - "/etc/systemd/system/weaveproxy.service"
+    - "/etc/systemd/system/weave.service"
+
+- name: Weave pre-upgrade | Reload systemd
+  command: systemctl daemon-reload
+  when: ansible_service_mgr == "systemd" and purged_weave_systemd_units.changed
```

Comment:
Every other systemctl call has it still. Consistency is better than "yeah, we don't need it still". I'll make a follow up for that.